src/test/codegen/issue-98678-native-enum.rs

```diff
@@ -4,12 +4,16 @@
 // compile-flags: -C debuginfo=2
 #![crate_type = "lib"]
 
-// CHECK: ![[#FILE:]] = !DIFile({{.*}}filename:{{.*}}src/test/codegen/issue-98678-native-enum.rs{{".*}})
+// NONMSVC: ![[#FILE:]] = !DIFile({{.*}}filename:{{.*}}src/test/codegen/issue-98678-native-enum.rs{{".*}})
+// MSVC: ![[#FILE:]] = !DIFile({{.*}}filename:{{.*}}src/test\\codegen\\issue-98678-native-enum.rs{{".*}})
 
-// CHECK: !DICompositeType({{.*"}}MyNativeEnum{{".*}}file: ![[#FILE]]{{.*}}line: [[# @LINE + 2]],
-// CHECK: !DICompositeType({{.*}}DW_TAG_variant_part{{.*}}file: ![[#FILE]]{{.*}}line: [[# @LINE + 1]],
+// NONMSVC: !DICompositeType({{.*"}}MyNativeEnum{{".*}}file: ![[#FILE]]{{.*}}line: [[# @LINE + 4]],
+// MSVC: !DICompositeType({{.*::}}MyNativeEnum>{{.*}}file: ![[#FILE]]{{.*}}line: [[# @LINE + 3]],
+// NONMSVC: !DICompositeType({{.*}}DW_TAG_variant_part{{.*}}file: ![[#FILE]]{{.*}}line: [[# @LINE + 2]],
+// COM: MSVC: currently no DW_TAG_variant_part from MSVC
```

Comment:
I think this might be because of https://github.com/rust-lang/rust/pull/104342/files#diff-a1ba92bf09005c702fd2e516c36d5660acea04dd3a8f871ff32bdbe478e6a034R363

Which reminds me, we should probably have test coverage for all of 

```rust
pub enum SingleCase {
    One,
}

pub enum MultipleDataCases {
    Case1(u32),
    Case2(i64),
}

pub enum NicheLayout {
    Something(&'static u32),
    Nothing,
}
```

Since all of these have different codepaths in debuginfo generation (at least, for `cpp_like` they do).